test-runner/src/main/scala/com/typesafe/spark/test/mesos/RolesSpec.scala

```diff
@@ -36,54 +35,13 @@ trait RoleSpecHelper {
 
     // pre-conditions
     if (cfg.role != "*") {
-      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role), "Spark role should exist.")
-    }
-
-    val expectedUsedCpus = {
-      val m = MesosCluster.loadStates(mesosConsoleUrl)
-      val tmp = m.slaves.map { x => x.resources.cpu }.sum
-      if (isInClusterMode) {
-          tmp - 1 // minus the cpus owned by the Spark Cluster, which is started before the tests in cluster mode
-      }
-      else tmp
+      assert(m.slaves.flatMap { x => x.reservedResources.keys }.contains(cfg.role), "Spark role should exist.")
     }
 
-    val mesosUrl = mesosConsoleUrl
-    val accum = sc.accumulator[Double](0L, "cpuCounter")
-    val partitions = 40 // give it enough tasks
-
-    // use enough tasks to utilize all expected resources for tested role and *, pick a task in the middle and observe resources
-    val rdd = sc.makeRDD(1 to 100, partitions).mapPartitionsWithIndex{ (idx, iterator) =>
-
-      var counter = 1
-      val retries = 1000
-      val partitionNumber = partitions / 2
-      val timeToSleep = 10 // small resolution to catch the event of a fully utilized cluster
-      var max = 0
-
-      if (idx == partitionNumber) {
-        while (counter <= retries) {
-          val currentNumOfCpus = MesosCluster.loadStates(mesosUrl).sparkFramework.get.resources.cpu
-
-          if (max <= currentNumOfCpus){ max = currentNumOfCpus.toInt }
-
-          if (currentNumOfCpus != expectedUsedCpus) {
-            Thread.sleep(timeToSleep)
-            counter += 1
-          } else {
-            counter = retries + 1 // exit
-          }
-        }
-
-        if (idx == partitionNumber) {
-          accum += max
-        }
-      }
-      iterator
-    }
+    assert(m.sparkFramework.get.role == cfg.role)
 
+    val rdd = sc.makeRDD(1 to 5)
     val res = rdd.sum()
-    assert(accum.value == expectedUsedCpus, "All cpu resources should be utilized.")
```

Comment:
Yet it does test it implicitly... :) if you check how resources are assigned in our run_test.sh script.

The tests are opinionated, Having the tests be generic to run in any cluster is a different requirement. It does not have to be that case in order to test roles.
The idea here is: if all cpus are used then i can deduce that role cpus were used, otherwise i will have to have only reserved resources for that role to test it isolated from the resources from default role(*) which will be offered if not eliminated. Right in our setup we needed to have (*) unreserved resources to run all the other tests which had no role along with the ones with a role. We use static resource allocation when a cluster is launched so we are a bit constrained here. Another solution would to use a different cluster launched for role tests after the first cluster is stopped running the non-role tests but that was extra work.